sdk/src/cache/messages/data/sorted_set/sorted_set_get_scores.rs

```diff
@@ -0,0 +1,239 @@
+        match value {
+            SortedSetGetScoresResponse::Hit {
+                values, responses, ..
+            } => {
+                let mut result = Vec::new();
+                for (bytes_value, response) in
+                    values.into_bytes().into_iter().zip(responses.into_iter())
+                {
+                    match response {
+                        SortedSetGetScoreResponse::Hit { score } => {
+                            let value: String = parse_string(bytes_value)?;
+                            let ele = SortedSetElement { score, value };
+                            result.push(ele);
+                        }
+                        SortedSetGetScoreResponse::Miss => (),
+                    }
+                }
+                Ok(result)
+            }
+            // In other SDKs we do not convert a `Miss` into an empty array
+            SortedSetGetScoresResponse::Miss => Err(MomentoError {
+                message: "sorted set get scores response was a miss".into(),
+                error_code: MomentoErrorCode::Miss,
+                inner_error: None,
+                details: None,
+            }),
+        }
+    }
+}
+
+/// Response for a sorted set get scores operation.
+///
+/// If you'd like to handle misses you can simply match and handle your response:
+/// ```
+/// # use momento::MomentoResult;
+/// use momento::cache::SortedSetGetScoresResponse;
+/// use std::convert::TryInto;
+/// # let response: SortedSetGetScoresResponse<Vec<String>> = SortedSetGetScoresResponse::Hit { responses: vec![], values: vec![] };
+/// match response {
+///     SortedSetGetScoresResponse::Hit { responses, values } => responses,
+///     SortedSetGetScoresResponse::Miss => return // probably you'll do something else here
+/// };
+/// ```
+///
+/// You can cast your result directly into a `Result<Vec<SortedSetElement<String>>, MomentoError>` suitable for
+/// ?-propagation if you know you are expecting a `Vec<SortedSetElement<String>>` item.
+///
+/// Of course, a Miss in this case will be turned into an Error. If that's what you want, then
+/// this is what you're after:
+/// ```
+/// # use momento::cache::{SortedSetGetScoresResponse, SortedSetElement};
+/// # use momento::MomentoResult;
+/// # let response: SortedSetGetScoresResponse<Vec<String>> = SortedSetGetScoresResponse::default();
+/// use std::convert::TryInto;
+/// let item: MomentoResult<Vec<SortedSetElement<String>>> = response.try_into();
+/// ```
+#[derive(Debug, PartialEq, PartialOrd)]
+pub enum SortedSetGetScoresResponse<T: IntoBytesIterable + Clone> {
+    /// The sorted set was found.
+    Hit {
+        /// The responses for each element.
+        responses: Vec<SortedSetGetScoreResponse>,
+        /// The values in the set that were requested
+        values: T,
+    },
+    /// The sorted set was not found.
+    Miss,
+}
+
+impl<T: IntoBytes + Clone> Default for SortedSetGetScoresResponse<Vec<T>> {
```

Comment:
I think we decided against implementing Default for the responses, especially if they'd be publicly visible and confusing to the user.

([comment for reference](https://github.com/momentohq/client-sdk-rust/pull/315#discussion_r1599129371))